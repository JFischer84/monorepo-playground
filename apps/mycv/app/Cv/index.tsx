import React from 'react';
import Image from 'next/image';
import {Accordion} from '@monorepo-playground/shared-components';
import styles from './index.module.scss';
import {ProgressBar} from '../ProgressBar';
import {Fact} from '../Fact';
import {Feat} from '../Feat';
import {AdventureLogEntry} from '../AdventureLogEntry';
import adventureLogEntries from '../adventureLogEntries';

interface CvProps {
  cvData: {
    name: string,
    classAndLevel: string,
    background: string,
    alignment: string,
    species: string
  },
  feats: string[]
}

export const Index = ({cvData, feats}: CvProps) => {
  const {name, classAndLevel, background, alignment, species} = cvData;

  const openLink = (link: string) => {
    window.open(link);
  };

  return (
    <div className={styles.cardContent}>
      <div className={styles.uperContainer}>
        <div className={styles.imageContainer}>
          <Image className={styles.profilePicture} width={256} height={256} src="/profile_pic2.jpg" alt="profile picture" />
        </div>
        <h3 className="card-title">{name}</h3>
        <div className={styles.factSection}>
          <div className={styles.factSubContainer}>
            <Fact factTitle={classAndLevel} description="Class & Level" />
            <Fact factTitle={background} description="Background" />
          </div>
          <div className={styles.factSubContainer}>
            <Fact factTitle={alignment} description="Alignment" />
            <Fact factTitle={species} description="Species" />
          </div>
        </div>
      </div>
      <div className={styles.progressBarContainer}>
        <Accordion title="Skills">
          <ProgressBar fillingPercentage="75" colorScheme="javascript" title="JavaScript" />
          <ProgressBar fillingPercentage="85" colorScheme="typescript" title="TypeScript" />
          <ProgressBar fillingPercentage="85" colorScheme="react" title="React" />
          <ProgressBar fillingPercentage="75" colorScheme="java" title="Java" />
          <ProgressBar fillingPercentage="60" colorScheme="springboot" title="Spring Boot" />
        </Accordion>
      </div>
      <div className={styles.featsContainer}>
        <Accordion title="Feats">
          <div className="row align-items-center gx-2">
            {feats.map((feat) => (
              <Feat key={feat} featTitle={feat} />
            ))}
          </div>
        </Accordion>
      </div>
      <div className={styles.adventureLogContainer}>
        <Accordion title="Adventure Log">
          <ul className="list-group list-group-flush">
            {adventureLogEntries.map((entry) => (
              <AdventureLogEntry key={entry.entryText} entry={entry} />
            ))}
          </ul>
        </Accordion>
      </div>
      <h4>Profiles</h4>
      <div className="row align-items-center gx-4">
        <div className="col">
          <button type="button" className="btn btn-primary" onClick={() => openLink('https://github.com/jfischer84')}>Github</button>
        </div>
        <div className="col">
          <button type="button" className="btn btn-primary" onClick={() => openLink('https://www.xing.com/profile/Jonas_Fischer34/cv')}>Xing</button>
        </div>
        <div className="col">
          <button type="button" className="btn btn-primary" onClick={() => openLink('https://www.linkedin.com/in/jonas-fischer-a5071998/')}>Linkedin</button>
        </div>
      </div>
    </div>
  );
};

export default Index;
