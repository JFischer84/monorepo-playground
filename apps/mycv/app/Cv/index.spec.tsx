import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Cv from './index';

const cvData = {name: 'Jonas Fischer',
  classAndLevel: 'Developer 7',
  background: 'Catlover/Gamer',
  alignment: 'Neutral Good',
  species: 'Human'};

const feats = ['cheese', 'chocolate', 'honey'];

describe('Cv', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Cv cvData={cvData} feats={feats} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
