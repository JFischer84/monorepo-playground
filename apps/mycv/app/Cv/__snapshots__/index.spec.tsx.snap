// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cv should render correctly 1`] = `
<div
  className="cardContent"
>
  <div
    className="uperContainer"
  >
    <div
      className="imageContainer"
    >
      <Image
        alt="profile picture"
        className="profilePicture"
        height={256}
        src="/profile_pic2.jpg"
        width={256}
      />
    </div>
    <h3
      className="card-title"
    >
      Jonas Fischer
    </h3>
    <div
      className="factSection"
    >
      <div
        className="factSubContainer"
      >
        <Fact
          description="Class & Level"
          factTitle="Developer 7"
        />
        <Fact
          description="Background"
          factTitle="Catlover/Gamer"
        />
      </div>
      <div
        className="factSubContainer"
      >
        <Fact
          description="Alignment"
          factTitle="Neutral Good"
        />
        <Fact
          description="Species"
          factTitle="Human"
        />
      </div>
    </div>
  </div>
  <div
    className="progressBarContainer"
  >
    <Accordion
      title="Skills"
    >
      <ProgressBar
        colorScheme="javascript"
        fillingPercentage="75"
        title="JavaScript"
      />
      <ProgressBar
        colorScheme="typescript"
        fillingPercentage="85"
        title="TypeScript"
      />
      <ProgressBar
        colorScheme="react"
        fillingPercentage="85"
        title="React"
      />
      <ProgressBar
        colorScheme="java"
        fillingPercentage="75"
        title="Java"
      />
      <ProgressBar
        colorScheme="springboot"
        fillingPercentage="60"
        title="Spring Boot"
      />
    </Accordion>
  </div>
  <div
    className="featsContainer"
  >
    <Accordion
      title="Feats"
    >
      <div
        className="row align-items-center gx-2"
      >
        <Feat
          featTitle="cheese"
          key="cheese"
        />
        <Feat
          featTitle="chocolate"
          key="chocolate"
        />
        <Feat
          featTitle="honey"
          key="honey"
        />
      </div>
    </Accordion>
  </div>
  <div
    className="adventureLogContainer"
  >
    <Accordion
      title="Adventure Log"
    >
      <ul
        className="list-group list-group-flush"
      >
        <AdventureLogEntry
          entry={
            Object {
              "entryText": "2018 - Today: Software Developer / Immowelt Hamburg GmbH",
              "isLast": true,
            }
          }
          key="2018 - Today: Software Developer / Immowelt Hamburg GmbH"
        />
        <AdventureLogEntry
          entry={
            Object {
              "entryText": "2015 - 2018: Software Developer Apprenticeship / Immowelt Hamburg GmbH",
            }
          }
          key="2015 - 2018: Software Developer Apprenticeship / Immowelt Hamburg GmbH"
        />
        <AdventureLogEntry
          entry={
            Object {
              "entryText": "2013 - 2015: Studies Game Programming / SAE Hamburg (Diploma)",
            }
          }
          key="2013 - 2015: Studies Game Programming / SAE Hamburg (Diploma)"
        />
        <AdventureLogEntry
          entry={
            Object {
              "entryText": "2012 - 2013: Nursery School Teacher Apprenticeship / Fachschule für Sozialpädagogik 2",
            }
          }
          key="2012 - 2013: Nursery School Teacher Apprenticeship / Fachschule für Sozialpädagogik 2"
        />
        <AdventureLogEntry
          entry={
            Object {
              "entryText": "2012 - 2012: Internships as Nursery School Teacher",
            }
          }
          key="2012 - 2012: Internships as Nursery School Teacher"
        />
        <AdventureLogEntry
          entry={
            Object {
              "entryText": "2011 - 2012: Call Center Agent / Axel Springer Direkt",
            }
          }
          key="2011 - 2012: Call Center Agent / Axel Springer Direkt"
        />
        <AdventureLogEntry
          entry={
            Object {
              "entryText": "2004 - 2011: Studies Media Culture / Universität Hamburg (Bakkalaureus Artium)",
            }
          }
          key="2004 - 2011: Studies Media Culture / Universität Hamburg (Bakkalaureus Artium)"
        />
      </ul>
    </Accordion>
  </div>
  <h4>
    Profiles
  </h4>
  <div
    className="row align-items-center gx-4"
  >
    <div
      className="col"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Github
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Xing
      </button>
    </div>
    <div
      className="col"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Linkedin
      </button>
    </div>
  </div>
</div>
`;
